components/GameScreen.vue

```diff
@@ -56,6 +63,80 @@ type DrawBotArgs = {
   graphics: Graphics;
 };
 
+function smoothZoom(Pos: { x: number; y: number }, scale: number) {
+  if (!appRef.value) {
+    return;
+  }
+
+  if (isZooming) {
+    return; // Prevent multiple simultaneous zooms
+  }
+  isZooming = true;
+  targetScale = scale;
+  targetPos = Pos;
+  startZoomTime = performance.now();
+
+  requestAnimationFrame(animateZoom);
+}
+
+function animateZoom(currentTime: number) {
+  if (!appRef.value || !appRef.value.stage || !startZoomTime) {
+    return;
+  }
+
+  const elapsedTime = currentTime - startZoomTime;
+  const progress = Math.min(elapsedTime / zoomDuration, 1);
+
+  // Get the current scale and the scale change based on progress
+  const currentScale = appRef.value.stage.scale.x;
+  const newScale = currentScale + (targetScale - currentScale) * progress;
+
+  // Calculate the world position relative to the current scale
+  const worldPos = {
+    x: (targetPos.x - appRef.value.stage.position.x) / appRef.value.stage.scale.x,
+    y: (targetPos.y - appRef.value.stage.position.y) / appRef.value.stage.scale.y,
+  };
+
+  appRef.value.stage.scale.set(newScale);
+
+  const newScreenPos = {
+    x: worldPos.x * newScale + appRef.value.stage.position.x,
+    y: worldPos.y * newScale + appRef.value.stage.position.y,
+  };
+
+  // Adjust the stage position to follow the zoom focus point
+  appRef.value.stage.position.set(
+    Math.min(0, Math.max(appRef.value.stage.position.x - (newScreenPos.x - targetPos.x), appRef.value.screen.width - appRef.value.screen.width * newScale)),
+    Math.min(0, Math.max(appRef.value.stage.position.y - (newScreenPos.y - targetPos.y), appRef.value.screen.height - appRef.value.screen.height * newScale)),
+  );
+
+  if (progress < 1) {
+    requestAnimationFrame(animateZoom);
+  } else {
+    isZooming = false;
+  }
+}
+
+function followPlayerBot(botx: number, boty: number, username: string) {
+  if (!appRef.value || !gameState.value) {
+    return;
```

Comment:
Should we `throw` here?

```suggestion
    throw new Error("unexpected: game not initialized");
```